server/sonar-server/src/main/java/org/sonar/server/platform/ServerFileSystemImpl.java

```diff
@@ -39,9 +42,9 @@
   private final File editionUninstallDir;
 
   public ServerFileSystemImpl(Configuration config) {
-    this.homeDir = new File(config.get(ProcessProperties.PATH_HOME).get());
-    this.tempDir = new File(config.get(ProcessProperties.PATH_TEMP).get());
-    this.dataDir = new File(config.get(ProcessProperties.PATH_DATA).get());
+    this.homeDir = new File(config.get(PATH_HOME.getKey()).get());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Call "Optional#isPresent()" before accessing the value. ([squid:S3655](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10300%2Fforbid_system_properties_in_settings_ws&issues=AWEnWsXngvt_XO9IEnRm&open=AWEnWsXngvt_XO9IEnRm)

[ISSUE_KEY]: AWEnWsXngvt_XO9IEnRm